### PR TITLE
pass vlan for mirror port

### PR DESCRIPTION
**Pull request by kthare10**
#343

### CODE REVIEW

File: fabrictestbed_extensions/fablib/interface.py [hunk 1]
```diff
@@ -623,19 +638,6 @@ def get_vlan(self) -> str:
...
-
-        :return: VLAN
-        :rtype: String
-        """
-        try:
-            bw = self.get_fim_interface().get_property(pname="capacities").bw
-        except:
-            bw = None
-        return bw
-
```
Comment (sajith) [inline fabrictestbed_extensions/fablib/interface.py:638]: Is this change intentional?  If so, since this is a public method, wouldn't it be better done in a separate change with some documentation (changelog, commit messages, etc) as to why this is being removed?

File: fabrictestbed_extensions/fablib/interface.py [hunk 2]
```diff
@@ -994,10 +996,11 @@ def config(self):
             # manual mode... do nothing
             pass
 
-    def add_mirror(self, port_name: str, name: str = "mirror"):
+    def add_mirror(self, port_name: str, name: str = "mirror", vlan: str = None):
```
Comment (sajith) [inline fabrictestbed_extensions/fablib/interface.py:999]: Would be helpful to have a docstring. :-)

File: fabrictestbed_extensions/fablib/interface.py [hunk 3]
```diff
@@ -341,6 +341,21 @@ def get_peer_port_name(self) -> str or None:
         else:
             return None
 
+    def get_peer_port_vlan(self) -> str:
+        """
+        Returns the VLAN associated with the interface.
+        For shared NICs extracts it from label_allocations.
+
+        :return VLAN to be used for Port Mirroring
```
Comment (sajith) [inline fabrictestbed_extensions/fablib/interface.py:349]: There's a missing `:` after `:return`.

**Review by sajith — APPROVED** [review]

File: fabrictestbed_extensions/fablib/interface.py (hunk 1 above)
Comment (kthare10) [reply fabrictestbed_extensions/fablib/interface.py:638]: There were multiple definitions for this function in this class. I removed one.

Comment (sajith) [reply fabrictestbed_extensions/fablib/interface.py:638]: Maybe there is a case for shadowing variables and methods in the same scope, but I wish Python would rather emit a warning in these cases...

Would have been useful to do this in a separate commit with a message, so that future readers won't have to puzzle over it. Oh well. :-)

Comment (kthare10) [reply fabrictestbed_extensions/fablib/interface.py:638]: I know but for the sake of ease, going to leave it here :)

Comment (sajith) [reply fabrictestbed_extensions/fablib/interface.py:638]: I see only one definition of `Interface.get_bandwidth()`. I must be missing something?

https://github.com/fabric-testbed/fabrictestbed-extensions/blob/46c4397a464d4add245f633d4dca6e4150bdce40/fabrictestbed_extensions/fablib/interface.py#L547

Also it seems that ruff or flake8 should detect some cases of redefintions.  Perhaps we could target some level of lint cleanliness.

https://docs.astral.sh/ruff/rules/redefined-while-unused/

Comment (kthare10) [reply fabrictestbed_extensions/fablib/interface.py:638]: May be that was fixed on main. All changes I have are based on rel1.7 which probably didn't have this fix.

Comment (kthare10) [reply fabrictestbed_extensions/fablib/interface.py:638]: rel1.7 branch has two definitions one at line 600 and another at 626

Comment (kthare10) [reply fabrictestbed_extensions/fablib/interface.py:638]: https://github.com/fabric-testbed/fabrictestbed-extensions/blob/6b2cc8dbeeadaa31d68bbc6bcb04c2d07771f004/fabrictestbed_extensions/fablib/interface.py#L600

https://github.com/fabric-testbed/fabrictestbed-extensions/blob/6b2cc8dbeeadaa31d68bbc6bcb04c2d07771f004/fabrictestbed_extensions/fablib/interface.py#L626

Comment (sajith) [reply fabrictestbed_extensions/fablib/interface.py:638]: Ah, thanks! I missed that this is being merged to rel1.7 branch.
